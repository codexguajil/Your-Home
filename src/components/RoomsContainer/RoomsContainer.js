import React, { Component } from 'react';
import Room from '../Room/Room'

export class RoomsContainer extends Component {
  
  renderRooms = () => {
    return this.props.rooms.map((room, i) => {
      return <Room key={i} {...room}/>
    });
  }

  render() {
    return (
      <div className='rooms'>
        {this.renderRooms()}
      </div>
    )
  }
}

export default RoomsContainer;