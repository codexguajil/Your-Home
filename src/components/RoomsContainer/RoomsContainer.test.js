import React from 'react';
import { shallow } from 'enzyme';
import RoomsContainer from './RoomsContainer';

describe('RoomsContainer', () => {
  let wrapper;
  let mockRooms;
  beforeEach(() => {
    mockRooms = [
      {
          "name": "Living Room 1",
          "type": "Living Room",
          "description": "Northeast living room",
          "roomMaterials": [
              {
                  "id": 1,
                  "element_type": "Flooring",
                  "material": {
                      "id": 1,
                      "name": "Material 1",
                      "model_number": "abc1",
                      "brand": "Kenmoore",
                      "vendor": "HD",
                      "manual_url": null,
                      "notes": null,
                      "quantity": null,
                      "unit_price": null
                  }
              }  
          ]
      },
      {
          "name": "Room 2",
          "type": "Kitchen",
          "description": "Big Kitchen",
          "roomMaterials": [
              {
                  "id": 3,
                  "element_type": "Shower",
                  "material": {
                      "id": 1,
                      "name": "Material 1",
                      "model_number": "abc1",
                      "brand": "Kenmoore",
                      "vendor": "HD",
                      "manual_url": null,
                      "notes": null,
                      "quantity": null,
                      "unit_price": null
                  }
              }
          ]
      }
  ]
    wrapper = shallow(<RoomsContainer rooms={mockRooms}/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the correct amount of rooms when renderRooms is called', () => {
    let result = wrapper.instance().renderRooms()
    expect(result.length).toBe(2)
  });

});