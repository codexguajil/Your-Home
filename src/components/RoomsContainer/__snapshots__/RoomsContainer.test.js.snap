// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomsContainer should match snapshot 1`] = `
<div
  className="rooms"
>
  <ConnectFunction
    description="Northeast living room"
    key="0"
    name="Living Room 1"
    roomMaterials={
      Array [
        Object {
          "element_type": "Flooring",
          "id": 1,
          "material": Object {
            "brand": "Kenmoore",
            "id": 1,
            "manual_url": null,
            "model_number": "abc1",
            "name": "Material 1",
            "notes": null,
            "quantity": null,
            "unit_price": null,
            "vendor": "HD",
          },
        },
      ]
    }
    type="Living Room"
  />
  <ConnectFunction
    description="Big Kitchen"
    key="1"
    name="Room 2"
    roomMaterials={
      Array [
        Object {
          "element_type": "Shower",
          "id": 3,
          "material": Object {
            "brand": "Kenmoore",
            "id": 1,
            "manual_url": null,
            "model_number": "abc1",
            "name": "Material 1",
            "notes": null,
            "quantity": null,
            "unit_price": null,
            "vendor": "HD",
          },
        },
      ]
    }
    type="Kitchen"
  />
</div>
`;
